mlflow/utils/virtualenv.py

```diff
@@ -254,10 +305,35 @@ def _execute_in_virtualenv(activate_cmd, command, install_mlflow, command_env=No
     :param install_mlflow: Flag to determine whether to install mlflow in the virtualenv
                            environment.
     :param command_env: Environment variables passed to a process running the command.
+    :param synchronous: Set the `synchronous` argument when calling `_exec_cmd`.
+    :param capture_output: Set the `capture_output` argument when calling `_exec_cmd`.
+    :param env_root_dir: See doc of PyFuncBackend constructor argument `env_root_dir`.
+    :param kwargs: Set the `kwargs` argument when calling `_exec_cmd`
     """
+    if command_env is None:
+        command_env = os.environ.copy()
+
+    if env_root_dir is not None:
+        command_env = {**command_env, **_get_virtualenv_extra_env_vars(env_root_dir)}
+
     pre_command = [activate_cmd]
     if install_mlflow:
         pre_command.append(_get_pip_install_mlflow())
+
+    if _IS_UNIX:
+        # Add "exec" before the starting scoring server command, so that the scoring server
+        # process replaces the bash process, otherwise the scoring server process is created
+        # as a child process of the bash process.
+        # Note we in `mlflow.pyfunc.spark_udf`, use prctl PR_SET_PDEATHSIG to ensure scoring
+        # server process being killed when UDF process exit. The PR_SET_PDEATHSIG can only
+        # send signal to the bash process, if the scoring server process is created as a
+        # child process of the bash process, then it cannot receive the signal sent by prctl.
+        # TODO: For Windows, there's no equivalent things of Unix shell's exec. Windows also
+        #  does not support prctl. We need to find an approach to address it.
+        command = "exec " + command
```

Comment:
Can we pull this out from this function?

Reply:
What's the function name for the pull out part do you suggest ?